internal/test/environment/testenv.go

```diff
@@ -14,7 +17,78 @@ func Setup() error {
 	if dockerHost == "" {
 		return errors.New("$TEST_DOCKER_HOST must be set")
 	}
-	return os.Setenv("DOCKER_HOST", dockerHost)
+	if err := os.Setenv("DOCKER_HOST", dockerHost); err != nil {
+		return err
+	}
+
+	if dockerCertPath := os.Getenv("TEST_DOCKER_CERT_PATH"); dockerCertPath != "" {
+		if err := os.Setenv("DOCKER_CERT_PATH", dockerCertPath); err != nil {
+			return err
+		}
+		if err := os.Setenv("DOCKER_TLS_VERIFY", "1"); err != nil {
+			return err
+		}
+	}
+
+	if kubeConfig := os.Getenv("TEST_KUBECONFIG"); kubeConfig != "" {
+		if err := os.Setenv("KUBECONFIG", kubeConfig); err != nil {
+			return err
+		}
+	}
+
+	if val := boolFromString(os.Getenv("TEST_REMOTE_DAEMON")); val {
+		if err := os.Setenv("REMOTE_DAEMON", "1"); err != nil {
+			return err
+		}
+	}
+
+	if val := boolFromString(os.Getenv("TEST_SKIP_PLUGIN_TESTS")); val {
+		if err := os.Setenv("SKIP_PLUGIN_TESTS", "1"); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// KubernetesEnabled returns if Kubernetes testing is enabled
+func KubernetesEnabled() bool {
+	return os.Getenv("KUBECONFIG") != ""
+}
+
+// SkipIfKubernetesDisabled skips the test if Kubernetes is not enabled
+func SkipIfKubernetesDisabled(t *testing.T) {
+	skip.If(t, !KubernetesEnabled(), "Kubernetes not enabled")
```

Comment:
These helpers aren't really necessary. It should already echo the second argument which reads basically the same way as the extra string.

If you keep them I would add `t.Helper()` here so at least it reports the correct line in the skip message.